samples/esb/esb_monitor/src/sniffer.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+
+#include <hal/nrf_radio.h>
+#include <esb.h>
+
+#include <SEGGER_RTT.h>
+
+#include "sniffer.h"
+
+LOG_MODULE_REGISTER(esb_sniffer, CONFIG_ESB_MONITOR_APP_LOG_LEVEL);
+
+static uint8_t rtt_data_buff[CONFIG_ESB_SNIFFER_RTT_DATA_BUFF_SZ];
+static uint8_t rtt_comm_ret_buff[CONFIG_ESB_SNIFFER_RTT_COMM_RET_BUFF_SZ];
+static uint8_t rtt_comm_buff[CONFIG_ESB_SNIFFER_RTT_COMM_BUFF_SZ];
+
+static k_tid_t rtt_comm_thread_id;
+static K_THREAD_STACK_DEFINE(rtt_comm_thread_stack, CONFIG_ESB_SNIFFER_THREAD_STACK_SZ);
+static struct k_thread rtt_comm_thread;
+
+static struct esb_sniffer_cfg sniffer_cfg;
+
+enum esb_sniffer_rtt_comm {
+	SNIFFER_START  = 1,
+	SNIFFER_STOP   = 2,
+	SNIFFER_STATUS = 3
+};
+
+static void radio_reg_to_array(uint32_t reg, char *array)
+{
+#if __CORTEX_M == (0x04U)
+	uint32_t inp = reg;
+
+	*((uint32_t *)array) = sys_cpu_to_be32((uint32_t)__RBIT(inp));
+#else
+	uint32_t inp = sys_cpu_to_le32(reg);
+
+	inp = (inp & 0xF0F0F0F0) >> 4 | (inp & 0x0F0F0F0F) << 4;
+	inp = (inp & 0xCCCCCCCC) >> 2 | (inp & 0x33333333) << 2;
+	inp = (inp & 0xAAAAAAAA) >> 1 | (inp & 0x55555555) << 1;
+	*((uint32_t *)array) =  inp;
```

Comment:
This can fail on misaligned buffer. Please use memcpy or similar to deal with potentially (32 bit) misaligned array pointers.